src/rev_ai/streamingclient.py

```diff
@@ -62,19 +62,24 @@ def __init__(self,
         self.on_connected = on_connected
         self.client = websocket.WebSocket(enable_multithread=True)
 
-    def start(self, generator, metadata=None):
+    def start(self, generator, custom_vocabulary_id=None, metadata=None):
         """Function to connect the websocket to the URL and start the response
             thread
 
         :param generator: generator object that yields binary audio data
         :param metadata: metadata to be attached to streaming job
         """
-        url = self.base_url + '?access_token={}'.format(self.access_token) \
-            + '&content_type={}'.format(self.config.get_content_type_string()) \
-            + '&user_agent={}'.format(quote('RevAi-PythonSDK/{}'.format(__version__), safe=''))
+        url = self.base_url + '?' + urlencode({
+            'access_token': self.access_token,
+            'content_type': self.config.get_content_type_string(),
+            'user_agent': 'RevAi-PythonSDK/{}'.format(__version__)
+        })
+
+        if custom_vocabulary_id:
+            url += '&' + urlencode({'custom_vocabulary_id': custom_vocabulary_id})
 
         if metadata:
-            url += '&metadata={}'.format(quote(metadata, safe=''))
+            url += '&' + urlencode({'metadata': metadata})
```

Comment:
Does this work with python 2.7 and python 3?

Reply:
I read this and it looks like urlencode is defined a little bit differently across python lang versions.
https://www.urlencoder.io/python/

Reply:
It should work in both python 2 and 3. This is why we have the try catch import statement at the top